### PR TITLE
run-local-server: Add "--results-dir" to use existing results

### PR DESCRIPTION
**Pull request by drothlis**
**TODO:**

* [x] Review the changes to `wait-for-socket.py` carefully, since the http-service startup depends on it.
* [x] Write some tests for `wait-for-socket.py`?
* [x] Test that http-service still starts on stb-tester ONE.

### CODE REVIEW

File: tests/test_utils.py [hunk 1]
```diff
@@ -96,7 +97,10 @@ def start(self, await_task_completion=True):
         with open('job-config.json', 'w') as f:
             f.write(json.dumps(self.job_config))
         self.process = subprocess.Popen(
-            ["%s/http-service/service.py" % SRC_DIR,
+            ["%s/wait-for-socket.py" % SRC_DIR,
+             "--needs-tcp-socket=localhost:%s" % self.mongodb_port,
+             "--",
+             "%s/http-service/service.py" % SRC_DIR,
```
Comment (wmanley) [inline tests/test_utils.py]: Why this change?  `mongod.start()` in `runner_ctx` should wait for MongoDB to be listening for connections before returning.

Comment (drothlis) [inline tests/test_utils.py]: `http-service/service.py` would get "connection refused" and thus fail to start.

**Comment by drothlis** [comment]
Pushed a new commit that removes the old buggy "wait for MongoDB to be listening" code from `test_utils.MongoDB.start` and replaces it with a correct implementation based on `wait-for-socket.py`.

TODO:

* [ ] Review the changes to `wait-for-socket.py` carefully, since the http-service startup depends on it.
* [ ] Write some tests for `wait-for-socket.py`?

File: tests/test_utils.py [hunk 2]
```diff
@@ -44,19 +44,22 @@ def __init__(self, tmpdir):
...
 
-        os.mkdir('%s/mongod_socket' % tmpdir)
-        os.mkdir('%s/mongodb' % tmpdir)
+        mkdir_p('%s/mongod_socket' % tmpdir)
+        mkdir_p('%s/mongodb' % tmpdir)
 
     def start(self):
+        wait_for_socket = imp.load_source(
+            "wait_for_socket",
+            os.path.join(os.path.dirname(__file__), "../wait-for-socket.py"))
```
Comment (wmanley) [inline tests/test_utils.py]: I'm not that fond of this. Is there another way using normal imports?

Comment (drothlis) [inline tests/test_utils.py]: I could pull the functionality out of `wait-for-socket.py` into `_stbt_service`... but then `wait-for-socket.py` would have to know where the `_stbt_service` python package is installed so we'd have to set PYTHONPATH.

Comment (drothlis) [inline tests/test_utils.py]: Nevermind, we already set PYTHONPATH in supervisord.conf.

**Comment by drothlis** [comment]
@wmanley This is ready for review now.
